Pytest: RuntimeError: wrapped C/C++ object of type ViewBox has been deleted

### Summary

I've seen this a few times recently in the pytest step on github actions.

```
view = <pyqtgraph.graphicsItems.ViewBox.ViewBox.ViewBox object at 0x7f2b62a070a0>

    def view_key(view):
>       return (view.window() is self.window(), view.name)
E       RuntimeError: wrapped C/C++ object of type ViewBox has been deleted
```


### Steps To Reproduce
Happens intermittently in automated testing. No reliable way to reproduce is known.

Probably related to garbage collection of the pyqtgraph ViewBox.

### Expected Behaviour

<!--- Explain how the program should behave once the issue has been resolved. -->

### Current Behaviour

<!--- Please provide information about the undesired current behaviour. -->

### Context

Seen today on Testing with CentOS 7 docker test
Its existed as far back as May 2023
I've seen it happen in a number of tests:
SpectrumWidgetTest.test_WHEN_reset_size_called_THEN_roi_size_reset
OperationsWindowsViewTest.test_lock_scale_changed_selected_not_flat_fielding
OperationsWindowsViewTest.test_lock_scale_changed_selected_flat_fielding
OperationsWindowsViewTest.test_collapse

### Failure Logs

```

=================================== FAILURES ===================================
___________________ OperationsWindowsViewTest.test_collapse ____________________
[gw0] linux -- Python 3.10.13 /opt/miniconda/envs/mantidimaging-dev/bin/python3.10

self = <mantidimaging.gui.windows.operations.test.view_test.OperationsWindowsViewTest testMethod=test_collapse>

    def setUp(self):
        # mock the view so it has the same methods
        with mock.patch("mantidimaging.gui.windows.main.view.WelcomeScreenPresenter"):
            self.main_window = MainWindowView()
>       self.window = FiltersWindowView(self.main_window)

mantidimaging/gui/windows/operations/test/view_test.py:22: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
mantidimaging/gui/windows/operations/view.py:85: in __init__
    self.previews = FilterPreviews(self)
mantidimaging/gui/windows/operations/filter_previews.py:55: in __init__
    self.imageview_difference = MIMiniImageView(name="difference", parent=self)
mantidimaging/gui/widgets/mi_mini_image_view/view.py:44: in __init__
    self.vb = ViewBox(invertY=True, lockAspect=True, name=name)
../miniconda/envs/mantidimaging-dev/lib/python3.10/site-packages/pyqtgraph/graphicsItems/ViewBox/ViewBox.py:228: in __init__
    self.register(name)
../miniconda/envs/mantidimaging-dev/lib/python3.10/site-packages/pyqtgraph/graphicsItems/ViewBox/ViewBox.py:286: in register
    ViewBox.updateAllViewLists()
../miniconda/envs/mantidimaging-dev/lib/python3.10/site-packages/pyqtgraph/graphicsItems/ViewBox/ViewBox.py:1738: in updateAllViewLists
    v.updateViewLists()
../miniconda/envs/mantidimaging-dev/lib/python3.10/site-packages/pyqtgraph/graphicsItems/ViewBox/ViewBox.py:1720: in updateViewLists
    nv = sorted(ViewBox.NamedViews.values(), key=view_key)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

view = <pyqtgraph.graphicsItems.ViewBox.ViewBox.ViewBox object at 0x7f2b62a070a0>

    def view_key(view):
>       return (view.window() is self.window(), view.name)
E       RuntimeError: wrapped C/C++ object of type ViewBox has been deleted

../miniconda/envs/mantidimaging-dev/lib/python3.10/site-packages/pyqtgraph/graphicsItems/ViewBox/ViewBox.py:1717: RuntimeError
```
From https://github.com/mantidproject/mantidimaging/actions/runs/7932046941/job/21657598087?pr=2040

--- a/docs/release_notes/next/fix-2053-viewbox-test-fail
+++ b/docs/release_notes/next/fix-2053-viewbox-test-fail
@@ -1,0 +1,1 @@
+#2053: Fix unreliable tests due to viewbox

--- a/mantidimaging/gui/widgets/mi_mini_image_view/view.py
+++ b/mantidimaging/gui/widgets/mi_mini_image_view/view.py
@@ -45,6 +45,7 @@ class MIMiniImageView(GraphicsLayout, BadDataOverlay, AutoColorMenu):
         self.vb.addItem(self.im)
         self.hist = HistogramLUTItem(self.im)
         graveyard.append(self.vb)
+        graveyard.append(self.hist.vb)
 
         # Sub-layout prevents resizing issues when details text changes
         image_layout = self.addLayout(colspan=2)

